File: java-streams/examples/java/src/main/java/streams/Filter.java
package streams;

import java.util.List;
import java.util.stream.Stream;

public class Filter {

    void examples(List<Person> persons) {

        Stream<Person> stream = persons.stream();

        Stream<Person> justBlackadder = stream
                .filter(person -> person.getLastName().equals("Blackadder"));

        Stream<Person> avoidFirstGeneration = stream
                .skip(5);

        Stream<Person> justFirstGeneration = stream
                .limit(5);

        Stream<Person> firstUnemployed = stream
                .takeWhile(person -> !person.isEmployed());// Java 9

        Stream<Person> skipFirstUnemployed = stream
                .dropWhile(person -> !person.isEmployed());// Java 9

        Stream<Person> skipUnemployedAndTakeWhileEmployed = stream
                .dropWhile(person -> !person.isEmployed()) // Java 9
                .takeWhile(person -> person.isEmployed());// Java 9
    }
}

File: java-streams/examples/kotlin/src/main/kotlin/streams/Filter.kt
package streams

class Filter {

    fun examples(persons: List<Person>) {

        val justBlackadder =
                persons.filter { it.lastName == "Blackadder" }

        val avoidFirstGeneration =
                persons.drop(5)

        val justFirstGeneration =
                persons.take(5)

        val firstUnemployed =
                persons.takeWhile { person -> !person.isEmployed }

        val skipFirstUnemployed =
                persons.dropWhile { person -> !person.isEmployed }

        val skipUnemployedAndTakeWhileEmployed = persons
                .dropWhile { person -> !person.isEmployed }
                .takeWhile { person -> person.isEmployed }
    }
}
